browser/net/ipfs_redirect_network_delegate_helper_browsertest.cc

```diff
@@ -112,4 +156,33 @@ IN_PROC_BROWSER_TEST_F(IpfsRedirectNetworkDelegateHelperBrowserTest,
   EXPECT_EQ(net::ERR_IPFS_DISABLED, observer.net_error_code());
 }
 
+IN_PROC_BROWSER_TEST_F(IpfsRedirectNetworkDelegateHelperBrowserTest,
+                       HTTPSSchemeDowngrade_CommonUrl) {
+  GetPrefs()->SetInteger(kIPFSResolveMethod,
+                         static_cast<int>(IPFSResolveMethodTypes::IPFS_LOCAL));
+
+  EXPECT_TRUE(
+      ui_test_utils::NavigateToURL(browser(), GURL("https://brave.com")));
+  EXPECT_EQ(web_contents()->GetURL(), GURL("https://brave.com"));
+}
+
+IN_PROC_BROWSER_TEST_F(IpfsRedirectNetworkDelegateHelperBrowserTest,
+                       HTTPSSchemeDowngrade_IPNSUrl) {
+  ResetTestServer(base::BindRepeating(
+      &IpfsRedirectNetworkDelegateHelperBrowserTest::HandleGetConnectedPeers,
+      base::Unretained(this)));
+
+  ipfs_service()->SetAllowIpfsLaunchForTest(true);
+
+  GetPrefs()->SetInteger(kIPFSResolveMethod,
+                         static_cast<int>(IPFSResolveMethodTypes::IPFS_LOCAL));
+
+  EXPECT_TRUE(ui_test_utils::NavigateToURL(
+      browser(),
+      GURL("https://en-wikipedia--on--ipfs-org.ipns.localhost:48080/wiki/")));
+  EXPECT_EQ(
+      web_contents()->GetURL(),
+      GURL("http://en-wikipedia--on--ipfs-org.ipns.localhost:48080/wiki/"));
+}
+
```

Comment:
maybe also add tests that `ipns://en.wikipedia-on-ipfs.org `, `http://localhost:48080/ipns/en.wikipedia-on-ipfs.org`, and`http://en.wikipedia-on-ipfs.org.ipns.localhost:48080/` end up going to `http://en-wikipedia--on--ipfs-org.ipns.localhost:48080/`?

Reply:
I've tried to do this, but there is a problem i've faced - BraveProxyingURLLoaderFactory's target_client_ is failing on the second redirect in browser tests, so only ipns->https or https->http, but not ipns->http requests are passing. I think this is compatibility issue between brave_request_handler and tests because this doesn't happen in the executable build.

Reply:
can you add the failing test as Disabled so we can look at it? It should be possible so not sure what's going on